server/src/metakb/query.py

```diff
@@ -613,46 +620,54 @@ def _get_disease(self, node: dict) -> MappableConcept:
-        :param relation: Relation type for categorical variant and variation
-        :return: List of variations with `relation` to categorical variant. If
-            VariationRelation.HAS_MEMBERS, returns at least one variation. Otherwise,
-            returns exactly one variation
-        """
-        query = f"""
-        MATCH (v:Variation) <- [:{relation.value}] - (cv:CategoricalVariant
-            {{ id: $cv_id }})
-        MATCH (loc:Location) <- [:HAS_LOCATION] - (v)
-        RETURN v, loc
+    @staticmethod
+    def _rebuild_allele(
+        allele_node: Node, location_node: Node, state_node: Node
+    ) -> Allele:
+        """Reconstruct allele from graph nodes
+
+        :param allele_node: allele node
+        :param location_node: location node
+        :param state_node: state node
+        :return: constructed VRS allele
         """
-        results = self.driver.execute_query(query, cv_id=cv_id).records
-        variations = []
-        for r in results:
-            r_params = r.data()
-            v_params = r_params["v"]
-            expressions = []
-            for variation_k, variation_v in list(v_params.items()):
-                if variation_k == "state":
-                    v_params[variation_k] = json.loads(variation_v)
-                elif variation_k.startswith("expression_hgvs_"):
-                    syntax = variation_k.split("expression_")[-1].replace("_", ".")
-                    expressions.extend(
-                        Expression(syntax=syntax, value=hgvs_expr)
-                        for hgvs_expr in variation_v
-                    )
-                    del v_params[variation_k]
-
-            v_params["expressions"] = expressions or None
-            loc_params = r_params["loc"]
-            v_params["location"] = loc_params
-            v_params["location"]["sequenceReference"] = json.loads(
-                loc_params["sequenceReference"]
+        if state_node.labels == {"SequenceExpression", "LiteralSequenceExpression"}:
+            state = LiteralSequenceExpression(sequence=state_node["sequence"])
+        elif state_node.labels == {"SequenceExpression", "ReferenceLengthExpression"}:
+            state = ReferenceLengthExpression(
+                length=state_node["length"],
+                repeatSubunitLength=state_node["repeat_subunit_length"],
+                sequence=state_node["sequence"],
             )
-            variations.append(Variation(**v_params).model_dump())
-        return variations
+        else:
+            msg = f"Unrecognized set of sequence expression node labels: {state_node.labels}"
+            raise ValueError(msg)
+
+        location = SequenceLocation(
+            start=location_node["start"],
+            end=location_node["end"],
+            id=location_node["id"],
+            sequenceReference={"refgetAccession": location_node["refget_accession"]},
+            sequence=location_node["sequence"],
+            digest=location_node.get("digest"),
```

Comment:
Maybe silly question. Are there cases where we wouldn't have a digest for a location?

Reply:
> FWIW, I would actually prefer to remove this property from the DB nodes, since it's effectively redundant information over the ID. I kept it in because it was gonna be effort to change the test cases and the tests were kinda driving me crazy.

This makes sense to me for locations. Wanna create an issue for this (if it doesn't already exist somewhere)?